core/Period/Range.php

```diff
@@ -587,4 +593,19 @@ public function getParentPeriodLabel()
     {
         return null;
     }
+
+    /**
+     * Returns the max allowed end timestamp for a range. If an enddate after this timestamp is provided, Matomo will
+     * automatically lower the end date to the date returned by this method.
+     * The max supported timestamp is always set to end of the current year plus 10 years.
+     *
+     * @return int
+     * @api
+     */
+    public static function getMaxAllowedEndTimestamp(): int
+    {
+        return strtotime(
+            date('Y-12-31 00:00:00', strtotime("+10 year", Date::$now ?? time()))
```

Comment:
```suggestion
            date('Y-12-31 00:00:00', strtotime('+10 year', Date::getNowTimestamp()))
```

Did not find any other direct usage of `Date::$now` outside of the tests, so we might just keep it like for a little while longer.